src/core/send.c

```diff
@@ -1042,37 +1081,10 @@ QuicSendPathChallenges(
             continue;
         }
 
-#ifdef QUIC_USE_RAW_DATAPATH
-        //
-        // Make sure the route is resolved before sending the path challenge.
-        //
-        // We need to set the path challenge flag back on so that when route is resolved,
-        // we know we need to continue to send the challenge.
-        //
-        CXPLAT_DBG_ASSERT(Path->Route.State != RouteSuspected);
-        if (Path->Route.State == RouteUnresolved) {
-            QuicConnAddRef(Connection, QUIC_CONN_REF_ROUTE);
-            QUIC_STATUS Status =
-                CxPlatResolveRoute(
-                    Path->Binding->Socket, &Path->Route, Path->ID, (void*)Connection, QuicConnQueueRouteCompletion);
-            if (Status == QUIC_STATUS_SUCCESS) {
-                QuicConnRelease(Connection, QUIC_CONN_REF_ROUTE);
-            } else {
-                //
-                // Route resolution failed or pended. We need to pause sending.
-                //
-                CXPLAT_DBG_ASSERT(Status == QUIC_STATUS_PENDING || QUIC_FAILED(Status));
-                Send->SendFlags |= QUIC_CONN_SEND_FLAG_PATH_CHALLENGE;
-                continue;
-            }
-        } else if (Path->Route.State == RouteResolving) {
-            //
-            // Can't send now. Once route resolution completes, we will resume sending.
-            //
+        if (CxPlatIsRawDatapath() && !CxPlatIsRouteReady(Connection, TRUE)) {
```

Comment:
We can actually do a tiny bit of refactoring and simplify this logic:

1. For socket-based datapaths, make `CxPlatResolveRoute` just set the route to the `RouteResolved` state inline, and then return success. There's nothing it needs to do because routes are practically already resolved always.
2. Remove the `CxPlatIsRawDatapath` function and just always call `CxPlatIsRouteReady`, which will call `CxPlatResolveRoute` if not already resolved.

Reply:
CxPlatIsRawDatapath can be removed from send.c, but it is needed for tests